### PR TITLE
Disable log function with config flag

### LINKED ISSUE #9939: log() function does not conform to SQL 2016
From section `6.30 <numeric value function>`:

```
<general logarithm function> ::=
  LOG <left paren> <general logarithm base> <comma>
      <general logarithm argument> <right paren>
```

The current implementation has the arguments reversed `log(value, base)`.

### PR DESCRIPTION
**Pull request by elonazoulay**
Add FeatureConfig flag to disable log and eventually remove it.
The 2 argument form of the log function does not conform to the latest sql standard.

Resolves #9939

### CODE REVIEW

**Comment by elonazoulay** [comment]
A few questions:
Did I put the test in the correct place?
Is there a less expensive/faster way to test?
Are there tests in presto-product-tests which might be affected (going through the instructions to run these in docker on laptop now)?

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java [hunk 1]
```diff
@@ -160,6 +160,19 @@ public static Slice absLong(@SqlType("decimal(p, s)") Slice arg)
...
     }
 
+    @ScalarFunction("log")
+    @Description("logarithm to given base")
+    public static final class LegacyLogFunction
+    {
+        private LegacyLogFunction() {}
+
+        @SqlType(StandardTypes.DOUBLE)
+        public static double log(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(StandardTypes.DOUBLE) double base)
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java]: `num` -> `number` (we don't use abbreviated variable names)

File: presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java [hunk 2]
```diff
@@ -153,6 +153,7 @@
     private long start;
     private ListeningExecutorService executor;
     private FunctionAssertions functionAssertions;
+
```
Comment (nezihyigitbasi) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java]: unnecessary change.

**Comment by nezihyigitbasi** [comment]
Please update the commit message title as `Add config flag to disable the log function`. Also the commit body says `flag to disable log by default`, but it's enabled by default.

**Comment by elonazoulay** [comment]
Thanks! Updating now.

File: presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            MaterializedResult result = queryRunner.execute(QUERY);
+            assertEquals(result.getOnlyValue(), 2.0);
+        }
+    }
+
+    @Test
+    public void testLegacyLogFunctionDisabled()
+    {
+        try (DistributedQueryRunner queryRunner = createQueryRunner(false)) {
+            MaterializedResult result = queryRunner.execute(QUERY);
```
Comment (nezihyigitbasi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java]: `result` is unused.

File: presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            fail("Expected exception");
+        }
+        catch (Exception e) {
+            assertContains(e.getMessage(), "Function log not registered");
+        }
+    }
+
+    public static DistributedQueryRunner createQueryRunner(boolean legacyLogFunction)
```
Comment (nezihyigitbasi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java]: this can be private

File: presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java [hunk 5]
```diff
@@ -597,6 +597,10 @@ public FunctionRegistry(TypeManager typeManager, BlockEncodingSerde blockEncodin
                 break;
         }
 
+        if (featuresConfig.isLegacyLogFunction()) {
+            builder.scalar(MathFunctions.LegacyLogFunction.class);
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java]: add an import for `LegacyLogFunction`.

File: presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    public void testLegacyLogFunctionDisabled()
+    {
+        try (DistributedQueryRunner queryRunner = createQueryRunner(false)) {
+            queryRunner.execute(QUERY);
+            fail("Expected exception");
+        }
+        catch (Exception e) {
+            assertContains(e.getMessage(), "Function log not registered");
```
Comment (findepi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java]: Since your try catch spans the whole test, it'd better with `@Test(expectedException=..., expectedExceptionMsg=..)`

However better yet would be to wrap just `queryRunner.execute(QUERY);` with a try-catch

File: presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }
+
+    private static DistributedQueryRunner createQueryRunner(boolean legacyLogFunction)
+            throws Exception
+    {
+        return DistributedQueryRunner.builder(testSessionBuilder().build())
+                .setNodeCount(1)
+                .setExtraProperties(legacyLogFunction ?
```
Comment (findepi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java]: just `ImmutableMap.of("deprecated.legacy-log-function", Boolean.toString(legacyLogFunction))` ?

File: presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        catch (Exception e) {
+            assertContains(e.getMessage(), "Function log not registered");
+        }
+    }
+
+    private static DistributedQueryRunner createQueryRunner(boolean legacyLogFunction)
+            throws Exception
+    {
+        return DistributedQueryRunner.builder(testSessionBuilder().build())
```
Comment (findepi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java]: Can you use `LocalQueryRunner`?

File: presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+public class TestLegacyLogFunction
+{
+    private static final String QUERY = "SELECT LOG(25, 5)";
+
+    @Test
+    public void testLegacyLogFunctionEnabled()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = createQueryRunner(true)) {
```
Comment (findepi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java]: `s/DistributedQueryRunner/QueryRunner` since the fact the runner is distributed is not important from test's perspective

(same in the other test method)

Comment (elonazoulay) [reply presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java]: Thanks a lot! Will update with your suggestions and add to my notes.
